datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -1240,7 +1294,10 @@ mod tests {
         std::mem::drop(output_stream0);
 
         // Now, start sending input
-        input.wait().await;
+        let mut background_task = JoinSet::new();
```

Comment:
What is the purpose of this change? I tried this change without the other changes in this PR and the test still passes (I was expecting it would hang or something)

Reply:
https://github.com/apache/arrow-datafusion/pull/10009#discussion_r1557396828